common/tools/dev-tool/src/commands/samples/checkNodeVersions.ts

```diff
@@ -67,20 +67,39 @@ async function cleanup(
   await deleteDockerImages(dockerImageNames);
 }
 
+function findSamplesDir(samplesDir: string, rootDir: string): string {
+  const dirs = [];
+  for (const file of fs.readdirSync(samplesDir)) {
+    const stats = fs.statSync(path.join(samplesDir, file));
+    if (stats.isDirectory()) {
+      const matchResult = file.match(/^v([0-9]*)$/);
+      if (matchResult !== null) {
```

Comment:
```suggestion
      if (/^v([0-9]*).*$/.test(file)) {
```

See the other comment about avoiding parseInt.

There are some problems with assuming that the dirs are always named `v${n}`, because they can be overwritten by the dev-tool scripts I think it's safe to assume they always _start_ with `v${n}`. For example, in Form Recognizer we had a `v3-beta` directory that was the most up-to-date version. Open to ideas about how to change this.

Reply:
Ah I see. Ok I updated it to check for a prefix with `v<numbers>.<anything>` then store the suffix without the `v` to be sorted later. I am not validating the anything part again a proper semvar syntax because i think it will be overkill for our purposes here.